src/main/java/io/cdap/plugin/transform/CreateRecordTransform.java

```diff
@@ -0,0 +1,448 @@
+    final Iterator<JsonElement> iterator = pathMap.iterator();
+    while (iterator.hasNext()) {
+      final JsonElement next = iterator.next();
+      if (isFirstValue) {
+        value = structuredRecord.get(next.getAsString());
+        isFirstValue = false;
+      } else {
+        value = ((StructuredRecord) value).get(next.getAsString());
+      }
+      // return null if one of the fields in path is not found
+      if (value == null) {
+        return null;
+      }
+    }
+    return value;
+  }
+
+  /**
+   * Get field from schema
+   *
+   * @param inputSchema {@link Schema} input schema
+   * @param pathMap     {@link JsonArray} path of the field to read
+   * @return return field found in given path
+   */
+  private Schema.Field getField(Schema inputSchema, JsonArray pathMap) {
+    Schema.Field value = null;
+    boolean isFirstValue = true;
+    final Iterator<JsonElement> iterator = pathMap.iterator();
+    while (iterator.hasNext()) {
+      final JsonElement next = iterator.next();
+      if (isFirstValue) {
+        value = inputSchema.getField(next.getAsString());
+        isFirstValue = false;
+      } else {
+        value = value.getSchema().getField(next.getAsString());
+      }
+      // return null if one of the fields in path is not found
+      if (value == null) {
+        return null;
+      }
+    }
+    return value;
+  }
+
+  /**
+   * Emit lineage
+   *
+   * @param context {@link StageSubmitterContext}
+   * @param schema  input schema
+   */
+  private void recordLineage(StageSubmitterContext context, Schema schema) {
+    if (schema == null) {
+      LOG.debug("The input schema is null. Field level lineage will not be recorded");
+      return;
+    }
+    JsonElement fieldMappingData = getFinalFieldMappingJson(schema);
+    LinkedList<FieldOperation> lineageData = generateLineageRecord("record", fieldMappingData.getAsJsonObject());
+    context.record(lineageData);
+  }
+
+  /**
+   * Generates lineage for each record in mapping json
+   *
+   * @param record
+   */
+  private LinkedList<FieldOperation> generateLineageRecord(String recordName, JsonElement record) {
+    LinkedList<FieldOperation> fieldOperations = new LinkedList<>();
+    List<String> inputFields = new ArrayList<>();
+    List<String> outputFields = new ArrayList<>();
+    for (Map.Entry<String, JsonElement> recordNode : record.getAsJsonObject().entrySet()) {
+      JsonElement value = recordNode.getValue();
+      if (value.isJsonArray()) {
+        // the field is not nested field - input is mapped to output directly
+        outputFields.add(recordNode.getKey());
+        inputFields.add(value.getAsJsonArray().get(0).getAsString());
+      }
+      if (value.isJsonObject()) {
+        // the field is nested object - make recursive call to map nested object fields
+        fieldOperations.addAll(generateLineageRecord(recordNode.getKey(), recordNode.getValue()));
+        outputFields.add(recordNode.getKey());
```

Comment:
We are adding an output field here, but no input field. Is that correct?

For example, if we have input table:
```
| field                 | type         |
| ------------          | ------------ |
| customer_profile      | Record       |
| - customer_name       | String       |
| - customer_phone      | String       |
```

with mapping:
```	
"customer": {
		"name": ["customer_profile", "customer_name"],
		"phone": ["customer_profile", "customer_phone"]
}
```

we would add a FieldTransOperation with inputFields=[], outputFields=["customer"].

Reply:
When it's nested record it calls recursive function (line 436) - which will hit condition on line 429 for fields inside the nested record and will get input and output fields and then create FieldTransformOperation in line 445 for the nested record. 

In the example you gave the "customer_profile" input field will be in the new FieldTransformOperation generated inside the recursive call.